src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisitionExtensionOptions.cs

```diff
@@ -28,5 +28,22 @@ internal void InvokeOnBeforeTokenAcquisitionForApp(AcquireTokenForClientParamete
                 OnBeforeTokenAcquisitionForApp(builder, acquireTokenOptions);
             }
         }
+
+        /// <summary>
+        /// Event fired when a ROPC flow request is being built.
+        /// </summary>        
+        public event BeforeTokenAcquisitionForUsernamePassword? OnBeforeTokenAcquisitionForUsernamePassword;
```

Comment:
Would it be possible to have a more general event `OnBeforeTokenAcquisitionForUser`
with an additional parameter in the delegate of type enum saying {UsernamePassword or webApp }
?

Reply:
Then how will the builder be passed to the delegate? Currently I am passing builder for the ROPC flow. [See ](https://github.com/AzureAD/microsoft-identity-web/pull/3130/files#diff-1bcbce3e8aad1539a2c83fba7250ba30c20daecaba15b31844f5d2178c2eef18R22)

Reply:
I don't think this can be made generic.

Reply:
Maybe Neha should add the `OnBefore` hooks for auth_code and for obo as well?

Reply:
@jmprieur could you please suggest?

Reply:
I can implement for the auth_code and obo as well if needed. I did not since those are not needed for now.

Reply:
To me this level of granularity that exists right now is fine and avoids having to define an enum. Semantically the 2 options are the same ... the grant needs to be explicit.

Reply:
I don't think we want a specific event like `BeforeTokenAcquisitionForUsernamePassword` but one for user and one for app w/the needed parameters.

would appreciate some more time to think this through.

Reply:
So the issue really is the name, I would like to not surface "UsernamePassword" in the extensibility options.

Reply:
I had a chat with @jmprieur and we'd like to propose a simple solution - instead of 

`public event BeforeTokenAcquisitionForUsernamePassword? OnBeforeTokenAcquisitionForUsernamePassword;` we will rename it all as 

`public event BeforeTokenAcquisitionForTestOnlyUser? OnBeforeTokenAcquisitionForTestOnlyUser;`

Reply:
Updated the name. Thanks!